client/src/components/RescheduleModal/RescheduleModal.jsx

```diff
@@ -33,29 +41,59 @@ export default class RescheduleModal extends React.PureComponent {
 
     _setSelectedSlot = (slot) => {
         const {bookingsListStore} = this.props;
-        bookingsListStore.setRescheduleModalData('selectedSlot', slot);
+        const {rescheduleModal} = bookingsListStore.store;
+        const {loading} = rescheduleModal;
+        if (!loading) {
+            bookingsListStore.setRescheduleModalData('selectedSlot', slot);
+        }
     };
 
     _renderSlotsSkeleton = () => {
         return (
             <div className={st(classes.slotsContainer)}>
-                {[...Array(MAX_SLOTS_AMOUNT)].map((_, index) => <RescheduleBoxSkeleton key={index}/>)}
+                {PARTS_OF_DAY.map((dayPart, index) => {
+                    return (
+                        <div key={index} className={st(classes.dayPart)}>
+                            <Text size="medium">{translate(`RescheduleModal.DayParts.${dayPart.name}`)}</Text>
+                            {range(dayPart.skeletonBoxes).map((_, index) => <RescheduleBoxSkeleton key={index}/>)}
+                        </div>
+                    );
+                })}
             </div>
         );
     };
 
+    _divideSlotsToDayParts = (slots) => {
+        return groupBy(slots, (slot) => {
+            const startHour = moment(slot.start.timestamp).hour();
+            if (startHour >= START_HOUR_OF_DAY && startHour < 12) {
+                return 'morning';
+            } else if (startHour >= 12 && startHour < 17) {
+                return 'afternoon';
+            } else if (startHour >= 17) {
+                return 'evening';
+            }
+        });
+    };
+
     _renderSlots = () => {
         const {bookingsListStore} = this.props;
         const {rescheduleModal} = bookingsListStore.store;
-        const {slots, selectedSlot} = rescheduleModal;
+        const {slots, selectedSlot, loading} = rescheduleModal;
+        const dividedSlots = this._divideSlotsToDayParts(slots);
         return (
             <div className={st(classes.slotsContainer)}>
-                {
-                    slots.slice(0, MAX_SLOTS_AMOUNT).map((slot, index) => (
-                            <RescheduleBox key={index} onClick={this._setSelectedSlot} isSelected={slot.clientId === (selectedSlot && selectedSlot.clientId)} data={slot}/>
-                        )
-                    )
-                }
+                {PARTS_OF_DAY.map((dayPart, index) => {
+                    return (
+                        <div key={index} className={st(classes.dayPart)}>
```

Comment:
just a heads up => it would have been better idea to use the name property ('morning', 'afternoon', 'evening') of the dayPart object.
imagine a scenario we were not rendering a day part which does not have available slots.
that could have mess up our rendering because of the usage with index.

for now you can leave it as is, it's just for general knowledge :)

I suggest you read this article as well.

[https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318](https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318)